sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/Readme.md

```diff
@@ -23,3 +44,616 @@ var client = new TimeSeriesInsightsClient(
+var hierarchySource = new TimeSeriesHierarchySource();
+hierarchySource.InstanceFieldNames.Add("hierarchyLevel1");
+
+var tsiHierarchy = new TimeSeriesHierarchy("sampleHierarchy", hierarchySource)
+{
+    Id = "sampleHierarchyId"
+};
+
+var timeSeriesHierarchies = new List<TimeSeriesHierarchy>
+{
+    tsiHierarchy
+};
+
+// Create Time Series hierarchies
+Response<TimeSeriesHierarchyOperationResult[]> createHierarchiesResult = await client
+    .Hierarchies
+    .CreateOrReplaceAsync(timeSeriesHierarchies)
+    .ConfigureAwait(false);
+
+// The response of calling the API contains a list of error objects corresponding by position to the input parameter array in the request.
+// If the error object is set to null, this means the operation was a success.
+for (int i = 0; i < createHierarchiesResult.Value.Length; i++)
+{
+    if (createHierarchiesResult.Value[i].Error == null)
+    {
+        Console.WriteLine($"Created Time Series hierarchy successfully.");
+    }
+    else
+    {
+        Console.WriteLine($"Failed to create a Time Series hierarchy: {createHierarchiesResult.Value[i].Error.Message}.");
+    }
+}
+```
+
+This code snippet demonstrates retrieving all hierarchies in your environment in pages. You can enumerate an AsyncPageable object using the `async foreach` loop.
+
+```C# Snippet:TimeSeriesInsightsSampleGetAllHierarchies
+// Get all Time Series hierarchies in the environment
+AsyncPageable<TimeSeriesHierarchy> getAllHierarchies = client.Hierarchies.GetAsync();
+await foreach (TimeSeriesHierarchy hierarchy in getAllHierarchies)
+{
+    Console.WriteLine($"Retrieved Time Series Insights hierarchy with Id: '{hierarchy.Id}' and Name: '{hierarchy.Name}'.");
+}
+```
+
+You can use a list of hierarchy IDs or names to get specific hierarchies, as demonstrated in this code snippet.
+
+```C# Snippet:TimeSeriesInsightsSampleGetHierarchiesById
+var tsiHierarchyIds = new List<string>
+{
+    "sampleHierarchyId"
+};
+
+Response<TimeSeriesHierarchyOperationResult[]> getHierarchiesByIdsResult = await client
+            .Hierarchies
+            .GetByIdAsync(tsiHierarchyIds)
+            .ConfigureAwait(false);
+
+// The response of calling the API contains a list of hieararchy or error objects corresponding by position to the input parameter array in the request.
+// If the error object is set to null, this means the operation was a success.
+for (int i = 0; i < getHierarchiesByIdsResult.Value.Length; i++)
+{
+    if (getHierarchiesByIdsResult.Value[i].Error == null)
+    {
+        Console.WriteLine($"Retrieved Time Series hieararchy with Id: '{getHierarchiesByIdsResult.Value[i].Hierarchy.Id}'.");
+    }
+    else
+    {
+        Console.WriteLine($"Failed to retrieve a Time Series hieararchy due to '{getHierarchiesByIdsResult.Value[i].Error.Message}'.");
+    }
+}
+```
+
+Similarly, you can use a list of hierarchies Ids or names to be able to delete heirarchies, as demonstrated in this code snippet.
```

Comment:
fixing misspell heirarchies

Reply:
Good catch. I have unconsciously ignored looking at spelling checks for  hierarchies. Adding a spell check extension in code editor would  be good idea.